crates/sui-types/src/gas_model/gas_v2.rs

```diff
@@ -0,0 +1,501 @@
+            object_read_per_byte_cost: c.obj_access_cost_read_per_byte(),
+            storage_per_byte_cost: c.obj_data_cost_refundable(),
+        }
+    }
+
+    pub(crate) fn unmetered() -> Self {
+        Self {
+            min_transaction_cost: 0,
+            max_gas_budget: u64::MAX,
+            package_publish_per_byte_cost: 0,
+            object_read_per_byte_cost: 0,
+            storage_per_byte_cost: 0,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SuiGasStatus<'a> {
+    // GasStatus as used by the VM, that is all the VM sees
+    gas_status: GasStatus<'a>,
+    // Cost table contains a set of constant/config for the gas model/charging
+    cost_table: SuiCostTable,
+    // Gas budget for this gas status instance.
+    // Typically the gas budget as defined in the `TransactionData::GasData`
+    gas_budget: u64,
+    // Computation cost after execution. This is the result of the gas used by the `GasStatus`
+    // properly bucketized.
+    // Starts at 0 and it is assigned in `bucketize_computation`.
+    computation_cost: u64,
+    // Whether to charge or go unmetered
+    charge: bool,
+    // Gas price for computation.
+    // This is a multiplier on the final charge as related to the RGP (reference gas price).
+    // Checked at signing: `gas_price >= reference_gas_price`
+    // and then conceptually
+    // `final_computation_cost = total_computation_cost * gas_price / reference_gas_price`
+    gas_price: u64,
+    // Gas price for storage. This is a multiplier on the final charge
+    // as related to the storage gas price defined in the system
+    // (`ProtocolConfig::storage_gas_price`).
+    // Conceptually, given a constant `obj_data_cost_refundable`
+    // (defined in `ProtocolConfig::obj_data_cost_refundable`)
+    // `total_storage_cost = storage_bytes * obj_data_cost_refundable`
+    // `final_storage_cost = total_storage_cost * storage_gas_price`
+    storage_gas_price: u64,
+    /// storage_cost is the total storage gas to charge. This is an accumulator computed
+    /// at the end of execution while determining storage charges.
+    /// It tracks `total_storage_cost = storage_bytes * obj_data_cost_refundable` as
+    /// described in `storage_gas_price`
+    /// It will be multiplied by the storage gas price.
+    storage_cost: u64,
+    /// storage_rebate is the total storage rebate (in Sui) accumulated in this transaction.
+    /// This is an accumulator computed at the end of execution while determining storage charges.
+    /// It is the sum of all `storage_rebate` of all objects mutated or deleted during
+    /// execution. The value is in Sui.
+    storage_rebate: u64,
+    // storage rebate rate as defined in the ProtocolConfig
+    rebate_rate: u64,
+    /// Amount of storage rebate accumulated when we are running in unmetered mode (i.e. system transaction).
+    /// This allows us to track how much storage rebate we need to retain in system transactions.
+    unmetered_storage_rebate: u64,
```

Comment:
This value seems never updated in this PR. Missed in the merge?

Reply:
possibly

Reply:
that was it! thanks!